### ISSUE TITLE
Trying to get a factory/service inside resource

### ISSUE THREAD

**Issue by luqezman** [zfcampus/zf-apigility#184]
Tell me one thing if you can help me:

How can I get a Factory/Service inside a resource?

**Comment by webimpress** [comment]
REST resource is created via container (service manager, if it is there defined) or directly:
https://github.com/zfcampus/zf-rest/blob/master/src/Factory/RestControllerFactory.php#L117-L121
If you want inject something intro REST resource you can define your own factory for the resource and inject whatever you want.

**Comment by luqezman** [comment]
Yes, but my ResourceFactory is not called or if I put a "exit" inside file, the php continues.

:'( I cloned apigility default project.

**Comment by luqezman** [comment]
I just cloned the project, I created the service in apigility admin but ResourceFactory is never called or used.

**Comment by webimpress** [comment]
Did you add to your configuration:
```php
'service_manager' => [
     'factories' => [
        YourApi\V1\Rest\YourService\Resource::class => YourResourceFactory::class,
    ],
],
```
?

You can use also abstract factory if you want, or the same factory for multiple resources.

**Comment by luqezman** [comment]
No I didn't. But I need to include this config inside module config or in global config?

**Comment by luqezman** [comment]
Yeah Nice, it works like a charm. Big thanks dude <3

**Comment by webimpress** [comment]
Doesn't matter. If this is just for one module it should be in module.config.php. If this is a global configuration for your application (applied for multiple modules) than it can be in global config.

**Comment by luqezman** [comment]
You can close this issue and a lot of people are searching for this in "google" or "stackoverflow".

Big thanks and HNY ;)

**Comment by weierophinney** [comment]
This repository has been closed and moved to laminas-api-tools/api-tools; a new issue has been opened at https://github.com/laminas-api-tools/api-tools/issues/10.
